builtin/providers/aws/resource_aws_db_option_group.go

```diff
@@ -0,0 +1,295 @@
+	} else {
+		d.Set("arn", arn)
+		resp, err := rdsconn.ListTagsForResource(&rds.ListTagsForResourceInput{
+			ResourceName: aws.String(arn),
+		})
+
+		if err != nil {
+			log.Printf("[DEBUG] Error retrieving tags for ARN: %s", arn)
+		}
+
+		var dt []*rds.Tag
+		if len(resp.TagList) > 0 {
+			dt = resp.TagList
+		}
+		d.Set("tags", tagsToMapRDS(dt))
+	}
+
+	return nil
+}
+
+func resourceAwsDbOptionGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	rdsconn := meta.(*AWSClient).rdsconn
+	if d.HasChange("option") {
+		o, n := d.GetChange("option")
+		if o == nil {
+			o = new(schema.Set)
+		}
+		if n == nil {
+			n = new(schema.Set)
+		}
+
+		os := o.(*schema.Set)
+		ns := n.(*schema.Set)
+		addOptions, addErr := expandOptionConfiguration(ns.Difference(os).List())
+		if addErr != nil {
+			return addErr
+		}
+
+		removeOptions, removeErr := flattenOptionConfigurationNames(os.Difference(ns).List())
+		if removeErr != nil {
+			return removeErr
+		}
+
+		modifyOpts := &rds.ModifyOptionGroupInput{
+			OptionGroupName: aws.String(d.Id()),
+		}
+
+		if len(addOptions) > 0 {
+			modifyOpts.OptionsToInclude = addOptions
+		}
+
+		if len(removeOptions) > 0 {
+			modifyOpts.OptionsToRemove = removeOptions
+		}
+
+		log.Printf("[DEBUG] Modify DB Option Group: %s", modifyOpts)
+		_, err := rdsconn.ModifyOptionGroup(modifyOpts)
+		if err != nil {
+			return fmt.Errorf("Error modifying DB Option Group: %s", err)
+		}
+
+	}
+
+	if arn, err := buildRDSOptionGroupARN(d, meta); err == nil {
+		if err := setTagsRDS(rdsconn, d, arn); err != nil {
+			return err
+		} else {
+			d.SetPartial("tags")
+		}
+	}
+
+	return resourceAwsDbOptionGroupRead(d, meta)
+}
+
+func resourceAwsDbOptionGroupDelete(d *schema.ResourceData, meta interface{}) error {
+	rdsconn := meta.(*AWSClient).rdsconn
+
+	deleteOpts := &rds.DeleteOptionGroupInput{
+		OptionGroupName: aws.String(d.Id()),
+	}
+
+	log.Printf("[DEBUG] Delete DB Option Group: %#v", deleteOpts)
+	_, err := rdsconn.DeleteOptionGroup(deleteOpts)
+	if err != nil {
+		return fmt.Errorf("Error Deleting DB Option Group: %s", err)
+	}
+
+	return nil
+}
+
+func flattenOptionConfigurationNames(configured []interface{}) ([]*string, error) {
+	var optionNames []*string
+	for _, pRaw := range configured {
+		data := pRaw.(map[string]interface{})
+		optionNames = append(optionNames, aws.String(data["option_name"].(string)))
+	}
+
+	return optionNames, nil
+}
+
+func resourceAwsDbOptionHash(v interface{}) int {
+	var buf bytes.Buffer
+	m := v.(map[string]interface{})
+	buf.WriteString(fmt.Sprintf("%s-", m["option_name"].(string)))
+	buf.WriteString(fmt.Sprintf("%d-", m["port"].(int)))
+
+	return hashcode.String(buf.String())
+}
+
+func buildRDSOptionGroupARN(d *schema.ResourceData, meta interface{}) (string, error) {
+	iamconn := meta.(*AWSClient).iamconn
+	region := meta.(*AWSClient).region
+	// An zero value GetUserInput{} defers to the currently logged in user
+	resp, err := iamconn.GetUser(&iam.GetUserInput{})
```

Comment:
We've historically had trouble with `GetUser` - it refers to the currently logged in user [unless we're authorized via an IAM instance profile](https://github.com/hashicorp/terraform/issues/3431), where it throws a `ValidationError`.

Is there a good reason to build the `arn` for this resource on behalf of the user? Looks to me like option group references elsewhere in the API all just use the name?